hudi-client/hudi-client-common/src/main/java/org/apache/hudi/execution/bulkinsert/BulkInsertSortMode.java

```diff
@@ -22,7 +22,8 @@
  * Bulk insert sort mode.
  */
 public enum BulkInsertSortMode {
-    NONE,
-    GLOBAL_SORT,
-    PARTITION_SORT
+  NONE,
+  GLOBAL_SORT,
+  PARTITION_SORT,
+  PARTITION_PATH_REDISTRIBUTE
```

Comment:
Redistribute while proper term isn't common in Spark glossary so might be confusing to users.

What do you think about `PARTITION_COLUMN_REPARTITION`? It's a bit of a tautology but relates directly to the repartitioning API Spark provides

Reply:
The reason to use `REDISTRIBUTE` is the new sort mode only does repartition for partitioned tables, not for non-partitioned tables, so I don't want to give the impression that the repartition is also done for non-partitioned tables.  But I understand your point that we'd like to call out repartition.  How about `PARTITION_PATH_SORT`?

Reply:
> not for non-partitioned tables, so I don't want to give the impression that the repartition is also done for non-partitioned tables.

I don't think there will be such confusion -- it's prefixed w/ `PARTITION_COLUMN` which doesn't exist for non-partitioned tables.

See, `PARTITION_PATH_SORT` leaves an impression that we sort by partition path, which is not what we're trying to do here.

Reply:
Makes sense.  As discussed, let's use `PARTITION_PATH_REPARTITION` as the name.